iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/json/ParameterDeclarationImpl.java

```diff
@@ -37,27 +37,27 @@
 public class ParameterDeclarationImpl extends AbstractArmTreeImpl implements ParameterDeclaration {
 
   private final Identifier identifier;
-  private final SimpleProperty type;
+  private final Property type;
   private final Property defaultValue;
   private final List<Expression> allowedValues;
-  private final SimpleProperty description;
-  private final SimpleProperty minValue;
-  private final SimpleProperty maxValue;
-  private final SimpleProperty minLength;
-  private final SimpleProperty maxLength;
+  private final Property description;
+  private final Property minValue;
+  private final Property maxValue;
+  private final Property minLength;
+  private final Property maxLength;
 
   // Methods should not have too many parameters
   @SuppressWarnings("java:S107")
   public ParameterDeclarationImpl(
     Identifier identifier,
-    SimpleProperty type,
+    Property type,
     @Nullable Property defaultValue,
     List<Expression> allowedValues,
-    @Nullable SimpleProperty description,
-    @Nullable SimpleProperty minValue,
-    @Nullable SimpleProperty maxValue,
-    @Nullable SimpleProperty minLength,
-    @Nullable SimpleProperty maxLength) {
+    @Nullable Property description,
+    @Nullable Property minValue,
+    @Nullable Property maxValue,
+    @Nullable Property minLength,
+    @Nullable Property maxLength) {
```

Comment:
You could be already more precise on the types. Eg. minLength will be a NumericLiteral. So in the end you don't need to do a type check as well as a type cast to simply access the value.